infra/gcp/README.md

```diff
@@ -0,0 +1,110 @@
+# infra/gcp
+
+Scripts and configuration files related to the management of the Kubernetes
+project's GCP infrastructure
+
+## layout
+
+Loosely organized as follows:
+- `bash/` to manage the bulk of our infrastructure (this came first)
+- `static/` static files deployed directly to our infrastructure
+- `terraform/` for managing the rest of our infrastructure
+
+The bash came first, and the terraform is slowly growing as our comfort with it
+grows. The decisions that led to some bash being in subdirectories were ad-hoc
+or seemed like good ideas at the time; consider them undocumented and open to
+reconsideration.
+
+A more detailed listing thanks to `tree`, with some files manually elided:
+```
+.
+├── bash
+│   ├── ensure-{foo}.sh     # entrypoints for managing the set of foo infrastructure
+│   ├── infra.yaml          # config for ensure-* scripts
+│   ├── lib.sh              # included by all ensure-* scripts
+│   ├── lib_{foo}.sh        # reusable bash functions for GCP service foo
+│   ├── backup_tools        # TODO: bash related to prod artifact hosting
+│   ├── cip-auditor         # TODO: bash related to prod artifact hosting
+│   ├── namespaces          # bash to manage namespaces / rbac for cluster: aaa
+│   ├── prow                # bash to manage k8s-infra-e2e* projects and prow secrets
+│   └── roles               # bash and yaml to manage custom GCP IAM roles
+├── static          # content related to prod artifact hosting
+└── terraform
+    ├── {foo}               # terraform managing resources in project foo
+    │   └── {bar}/resources # k8s resources deployed to GKE cluster {bar}
+    └── modules             # modules for re-use within this repo
+```
+
+## bash
+
+### deploying changes
+
+The bash in here needs to be manually run by humans. At the moment, this is
+usually restricted to members of k8s-infra-gcp-org-admins@kubernetes.io
+
+### principles
```

Comment:
@spiffxp my life goal is to reach your level of communication skills. :-)

Reply:
I'll leave the README here for now then